### PR TITLE
Move Protocol Buffers to Common Directory

### PR DESCRIPTION
**Pull request by chases2**
It's really confusing that there's 4 directories with important-sounding names like `config` and `summary` that just lead to .proto files

To do:
- [x] readme link changes

### CODE REVIEW

**Comment by fejta** [comment]
I think this is gross, package names should not have protobuf in them

**Comment by chases2** [comment]
> I think this is gross, package names should not have protobuf in them

Why not? Is there a better name for the directory that contains Testgrid-wide data structures?

**Comment by fejta** [comment]
Where are all our packages named yaml and json?

**Comment by chases2** [comment]
`s/protobuf/data/g`

@fejta I think this meets your requirements because it identifies the commonality of these as being Testgrid-specific data structures, without specifying that they are necessarily protocol buffers. The name describes function, not data type.

Third choice was `struct`, but I think that's slightly less descriptive.

File: testgrid/cmd/configurator/yaml2proto.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ func (c *Config) MarshalText(w io.Writer) error {
 	return proto.MarshalText(w, c.config)
 }
 
-// MarshalBytes returns the wire-encoded protobuf data for the parsed configuration.
+// MarshalBytes returns the wire-encoded data data for the parsed configuration.
```
Comment (fejta) [inline testgrid/cmd/configurator/yaml2proto.go]: data data?

**Comment by chases2** [comment]
I'm fine with keeping the files where they are. 

_If_ they're kept where they are, though, then the refactoring I'm doing as part of #13690 is going to add more code to the config file ([yaml2proto.go](https://github.com/kubernetes/test-infra/blob/master/testgrid/cmd/configurator/yaml2proto.go) belongs there at least).

I assumed that there was a desire to keep these protodef folders containing only protos, but if that's not a thing we care about, then I'll close this PR.

**Comment by fejta** [comment]
Right now they don't only contain protos, so this isn't something I would say is important.

**Pull request closed without merging**